pkg/virt-operator/creation/components/crds.go

```diff
@@ -327,7 +327,9 @@ func NewPrometheusRuleSpec(ns string) *promv1.PrometheusRuleSpec {
 					},
 					{
 						Record: "num_of_allocatable_nodes",
-						Expr:   intstr.FromString("count(count (kube_node_status_allocatable) by (node))"),
+						// Since we don't set any specific node label selector, any node with at least 1 NoSchedule
+						// effect will prevent our workloads from being scheduled on that node.
+						Expr: intstr.FromString("count(count (kube_node_status_allocatable) by (node)) - count(kube_node_spec_taint{effect='NoSchedule'})"),
```

Comment:
I think a node can have multiple NoSchedule taints. I wonder how accurate this is. Is it possible to only count nodes without any "NoSchedule" taint? 

Also, is it not possible that an allocatable node having a "NoSchedule" should actually result in alerts being fired if it disrupts our component availability ?

Reply:
> I think a node can have multiple NoSchedule taints. I wonder how accurate this is. Is it possible to only count nodes without any "NoSchedule" taint?

I don't think that we can query Prometheus for nodes without the `taint` because if `taint` was not set, `kube-state-metrics` will just not expose a metric for it: https://github.com/kubernetes/kube-state-metrics/blob/f8bcc024791126e1a8b3f7d07e4392a5373b66e7/internal/store/node.go#L150

> Also, is it not possible that an allocatable node having a "NoSchedule" should actually result in alerts being fired if it disrupts our component availability ?

I'm starting to think that we should (or maybe even can't) precisely decide from the metrics if a pod (for example: virt-launcher) can be placed on a node because there is more than just tolerations that can affect that decision. I suggest to have just an alert for missing pods from a DaemonSet (x out of y desired) and once this alert fires, it's up to the cluster admin to check what is wrong with his cluster instead of us trying to predict something that is not 100% accurate and can mislead.

@davidvossel what do you think?